lib/appendGitlabProv.js

```diff
@@ -0,0 +1,72 @@
+import rdf from 'rdf-ext'
+import { Transform } from 'readable-stream'
+import { checkGitlabVars, provFromGitlab } from './metadata/produceProv.js'
+import { prov } from './namespaces.js'
+
+const typePredicate = rdf.namedNode(
+  'http://www.w3.org/1999/02/22-rdf-syntax-ns#type')
+
+class ProvMetadata extends Transform {
+  constructor (context, { subjectsWithClass, graph }) {
+    super({ objectMode: true })
+    this.type = subjectsWithClass
+    const { omitProv, message } = checkGitlabVars()
+    if (omitProv) {
+      console.warn(message)
```

Comment:
Isn't console caught by the linter? The context has a logger:

```suggestion
      context.logger.warn(message)
```